src/backend/distributed/commands/multi_copy.c

```diff
@@ -3509,6 +3555,51 @@ CopyGetPlacementConnection(ShardPlacement *placement, bool stopOnFailure)
 }
 
 
+static MultiConnection *
+GetConnectionIfExecutorPoolSizeReached(HTAB *connectionStateHash, char *nodeName, int
```

Comment:
I think it'd be nicer to move the if statement to the caller and give this function a name that describes what it does, e.g. GetLeastUtilisedCopyConnection

Reply:
renamed to `GetLeastUtilisedCopyConnection` as suggested